src/components/SchemaViewer.svelte

```diff
@@ -1,11 +1,20 @@
 <script>
+  import page from "page";
   import SchemaNode from "./SchemaNode.svelte";
   import FilterInput from "./FilterInput.svelte";
 
   export let app;
   export let nodes = [];
+  export let queryString;
   let nodesWithVisibility;
-  const filterTextChanged = (filterText = "") => {
+
+  function updateQueryString(filterText) {
+    queryString = filterText;
+    page(`${window.location.hash.split("?")[0]}?search=${queryString}`);
```

Comment:
This works, but doesn't seem quite right to me. Any suggestions on how this could be done in a better way?

Reply:
Hmm, I think we should probably define a callback function inside the components using this, and delegate the responsibility of updating the URL to them. Calling into page.js from a sub-component seems like a pretty big violation of seperation-of-concerns.

After that, we should be able to pass this page-level data to the component via page.js's callback facility, which is a little cleaner than what you're doing here. However, that's a little more involved, I think for a first pass what you're doing is fine-- we can do that fairly easily in a followup later.